docs/the-new-architecture/pillars-codegen.md

```diff
@@ -3,5 +3,227 @@ id: pillars-codegen
+:::
+
+Then, we need to add the module that requires the **CodeGen** as an npm dependency of the app. This can be done with the command:
+
+```sh
+yarn add <path/to/your/TurboModule_or_FabricComponent>
+```
+
+See how to create a [TurboModule](./pillars-turbomodule) or a [Fabric Component](./pillar-fabric-components) to get more information on how to configure them.
+
+The rest of this guide assumes that you have a `TurboModule` and/or a `Fabric Component` properly set up.
+
+# iOS
+
+## Running the CodeGen
+
+The **CodeGen** for iOS relies on some Node scripts that are invoked during the build process. The scripts are located in the `MyApp/node_modules/react_native/scripts/` folder.
+
+The script to run is the `generate-artifacts.js` script. This searches among all the dependencies of the app, looking for JS files which respects some specific conventions (look at [TurboModules](./pillars-turbomodule) and [Fabric Components](./pillar-fabric-components) sections for details) and generates the required code.
+
+To invoke the script we can run this command from the root folder of our app:
+
+```sh
+node node_modules/react_native/scripts/generate-artifacts.js \
+    --path MyApp/ \
+    --outputPath <an/output/path> \
+```
+
+Given that the app has `TurboModules` and/or `Fabric Components` configured as a dependency, the **CodeGen** will look for all of them and it will generate the code in the `an/outputh/path` path.
+
+## The Generated Code
+
+If we run the **CodeGen** in our app with an output path of `codegen`, we obtain the following structure:
+
+```sh title="iOS CodeGen output"
+codegen
+└── build
+    └── generated
+        └── ios
+            ├── MyTurboModuleSpecs
+            │   ├── MyTurboModuleSpecs-generated.mm
+            │   └── MyTurboModuleSpecs.h
+            ├── FBReactNativeSpec
+            │   ├── FBReactNativeSpec-generated.mm
+            │   └── FBReactNativeSpec.h
+            ├── RCTThirdPartyFabricComponentsProvider.h
+            ├── RCTThirdPartyFabricComponentsProvider.mm
+            └── react
+                └── renderer
+                    └── components
+                        ├── MyFabricComponent
+                        │   ├── ComponentDescriptors.h
+                        │   ├── EventEmitters.cpp
+                        │   ├── EventEmitters.h
+                        │   ├── Props.cpp
+                        │   ├── Props.h
+                        │   ├── RCTComponentViewHelpers.h
+                        │   ├── ShadowNodes.cpp
+                        │   └── ShadowNodes.h
+                        └── rncore
+                            ├── ComponentDescriptors.h
+                            ├── EventEmitters.cpp
+                            ├── EventEmitters.h
+                            ├── Props.cpp
+                            ├── Props.h
+                            ├── RCTComponentViewHelpers.h
+                            ├── ShadowNodes.cpp
+                            └── ShadowNodes.h
```

Comment:
This should not be here though 🤔

Reply:
Why not?

Reply:
On Android, `rncore` is generated only once as part of ReactAndroid. It won't be generated as part of users' codegen invocation